packages/core/src/DayOfWeek.js

```diff
@@ -93,7 +93,7 @@ export class DayOfWeek extends TemporalAccessor {
      */
     static of(dayOfWeek) {
         if (dayOfWeek < 1 || dayOfWeek > 7) {
-            throw new DateTimeException('Invalid value for DayOfWeek: ' + dayOfWeek);
+            throw new DateTimeException('Invalid value for DayOfWeek: ' + dayOfWeek.toString());
```

Comment:
Why add a .toString() here? Is the method implemented for all types ? And isn't toString used anyway ?

Reply:
Indeed, this is not needed. I put it because I was first thinking in implement this method for every major type; at the end I just make a first experiment only with `Temporal` to see how it goes and discuss about it.

What do you think? What types should have the same behavior? Only `Temporal` and `TemporalAmount`? All of them?

Reply:
Not sure if i get your answer right. The `.toString()` is not required, because `toString()` method that is automatically called when the object is to be represented as a text value. And it is implemented for all threeten types.
And yes, IMO it would be nice to convert to string literals, whenever we touch a piece of code. so ``Invalid value for DayOfWeek: ${dayOfWeek}`` should do the trick

Reply:
And yes, i would say we should also throw in `TemporalAmount.valueOf`

Reply:
Sorry I wasn't clear. In the expression `"text" + o` where `o` is an object, the method `o.valueOf()` is called even if the LHS is a string; `.valueOf()` defaults to `.toString()` if isn't implemented (there is also `Symbol.toPrimitive` but let us left that aside for now). Even if an object has an implementation for `.toString()` it is not called if there is an implementation for `.valueOf()`. OTOH a template literal string does call `.toString()` first and not `.valueOf()`.

When I started this I was thinking to implement a throwing `valueOf` for almost every type, but then just did for `Temporal` to see how it goes. This particular line was the "fix" for a `.valueOf()` that throws in `DayOfWeek` but at the end I submit a draft PR to discuss about this before implementing more. If we decided that `DayOfWeek` won't have that method then I will revert this line, if it should throw as well then this `.toString()` is necessary (or changing it to a template literal).

Reply:
Ok, that changes things a bit. That would mean, this PR breaks a lot, correct ?

A `console.log('Today is ' + LocalDate.now());` would then throw an exception, because `valueOf()` is called internally. In opposite ``console.log(`Today is ${LocalDate.now()}`);`` would not throw, because it fallbacks to `toString()`

Hmm, i have to think about it, not sure if this is what we want ? 

I just checked temporal/proposal polyfill, they have the same issue. `console.log('foo' + Temporal.Now.plainDateISO());` throws a TypeError.

Reply:
@phueper any opinion on that ? 

In general it's about: throwing a Typeerror on any comparison via operators, so that a user can't perform an invalid comparison, eg `LocalDate.of(2017,1,10)  < LocalDate.of(2017,1,10)` would throw instead of returning false. 

In consequence that is a major change, that disables String concatenation as well.

Reply:
my thoughts:
- in general i do like the idea of throwing for example when comparing
- throwing when doing `console.log('Date:' + LocalDate.now())` however seems a little risky to me .. i would expect a lot of existing code to break with that ( i can remember me doing this several times :)) but i assume there is no way for us to know wether valueOf is called for this kind of string conversion or for another operation?

Reply:
Is there an autofix option for string literals in eslint ?